nixos/modules/system/boot/timesyncd.nix

```diff
@@ -46,6 +46,29 @@ with lib;
       wantedBy = [ "sysinit.target" ];
       aliases = [ "dbus-org.freedesktop.timesync1.service" ];
       restartTriggers = [ config.environment.etc."systemd/timesyncd.conf".source ];
+
+      preStart = let
```

Comment:
`preStart` is to `serviceConfig.ExecStartPre` as `script` is to `serviceConfig.ExecStart`. You don't need an extra writeScript here :)